src/app-layout/visual-refresh-toolbar/drawer/global-drawer.tsx

```diff
@@ -0,0 +1,137 @@
+  const prevShow = usePrevious(show);
+
+  useEffect(() => {
+    if (activeGlobalDrawer?.preserveInactiveContent && !!prevShow !== show) {
+      if (show) {
+        fireNonCancelableEvent(activeGlobalDrawer?.onShow);
+      } else {
+        fireNonCancelableEvent(activeGlobalDrawer?.onHide);
+      }
+    }
+  }, [activeGlobalDrawer, prevShow, show]);
+
+  const computedAriaLabels = {
+    closeButton: activeGlobalDrawer ? activeGlobalDrawer.ariaLabels?.closeButton : ariaLabels?.toolsClose,
+    content: activeGlobalDrawer ? activeGlobalDrawer.ariaLabels?.drawerName : ariaLabels?.tools,
+  };
+
+  const activeDrawerSize = activeGlobalDrawersSizes[activeDrawerId ?? ''] ?? 0;
+  const minDrawerSize = minGlobalDrawersSizes[activeDrawerId ?? ''] ?? 0;
+  const maxDrawerSize = maxGlobalDrawersSizes[activeDrawerId ?? ''] ?? 0;
+  const refs = globalDrawersFocusControl.refs[activeDrawerId];
+  const resizeProps = useResize({
+    currentWidth: activeDrawerSize,
+    minWidth: minDrawerSize,
+    maxWidth: maxDrawerSize,
+    panelRef: drawerRef,
+    handleRef: refs?.slider,
+    onResize: size => onActiveDrawerResize({ id: activeDrawerId!, size }),
+  });
+  const size = getLimitedValue(minDrawerSize, activeDrawerSize, maxDrawerSize);
+  const lastOpenedDrawerId = drawersOpenQueue.length ? drawersOpenQueue[0] : null;
+
+  return (
+    <aside
+      id={activeDrawerId}
+      aria-hidden={!show}
+      aria-label={computedAriaLabels.content}
+      className={clsx(styles.drawer, sharedStyles['with-motion'], {
+        [styles['drawer-global']]: show,
+        [styles['last-opened']]: lastOpenedDrawerId === activeDrawerId,
+        [testutilStyles['active-drawer']]: show,
+      })}
+      ref={drawerRef}
+      onBlur={e => {
+        if (!e.relatedTarget || !e.currentTarget.contains(e.relatedTarget)) {
+          globalDrawersFocusControl.loseFocus();
+        }
+      }}
+      style={{
+        blockSize: show ? `calc(100vh - ${placement.insetBlockStart}px - ${placement.insetBlockEnd}px)` : 0,
+        insetBlockStart: placement.insetBlockStart,
+        ...(!isMobile && { [customCssProps.drawerSize]: `${show ? size : 0}px` }),
+      }}
+      data-testid={`awsui-app-layout-drawer-${activeDrawerId}`}
+    >
+      {!isMobile && activeGlobalDrawer?.resizable && (
+        <div className={styles['drawer-slider']}>
+          <PanelResizeHandle
+            ref={refs?.slider}
+            position="side"
+            className={testutilStyles['drawers-slider']}
+            ariaLabel={activeGlobalDrawer?.ariaLabels?.resizeHandle}
+            ariaValuenow={resizeProps.relativeSize}
+            onKeyDown={resizeProps.onKeyDown}
+            onPointerDown={resizeProps.onPointerDown}
+            compact={true}
+          />
+        </div>
+      )}
+      <div className={clsx(styles['drawer-content-container'], sharedStyles['with-motion'])}>
+        <div className={clsx(styles['drawer-close-button'])}>
```

Comment:
Instead of copying this HTML structure, should we create a `<BaseDrawer>` component? How much common HTML can we extract?

Reply:
I don't think that creating such a component brings much benefits, because there are multiple conditions for styles / handlers / content